### PR TITLE
HS-1783, HS-1772: Store alertable events in alert service database

### PR DESCRIPTION
**Pull request by geraldhumphries**
## Description

- Refactors alert service to store its list of alertable events as an entity called EventDefinition (final name TBD)
- Adds API for EventDefinition
- Queries for EventDefinitions on the frontend and uses them in the alert condition form
- EventType enum has been repurposed to specify "SNMP trap", "internal event", etc.

Todo:

- [ ] Rename EventDefinition to something better like AlertableEvent
- [x] Sort out type names in gRPC API vs rest of codebase (e.g. entities)
- [x] Fix `watch` not working properly in AlertCondition form
- [ ] Sort out the EventType input on the PolicyRule form

## Jira link(s)
- https://opennms.atlassian.net/browse/HS-1783
- https://opennms.atlassian.net/browse/HS-1772

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesCard.vue [hunk 1]
```diff
@@ -54,13 +54,18 @@
...
-        <MonitoringPoliciesCardAlertRow
-          v-for="(condition, index) in rule.alertConditions"
-          :key="condition.id"
-          :rule="rule"
-          :condition="condition"
-          :index="index"
+        <MonitoringPoliciesCardEventConditionTable
+          :eventConditions="rule.alertConditions"
         />
+<!--    TODO: Bring this back when thresholding is implemented
```
Comment (scthompsonopen) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesCard.vue:60]: I would recommend creating a JIRA for this TODO so we can track it properly.

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue [hunk 2]
```diff
@@ -94,7 +94,8 @@
                 :condition="(cond as EventCondition)"
                 :policy="store.selectedPolicy"
                 :rule="store.selectedRule"
-                :index="index"
+                :event-type="EventType.SnmpTrap"
+                :index="index as number"
```
Comment (scthompsonopen) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue:98]: Same comment as above, both are nits and possible to ignore in a time crunch though.

**Review by scthompsonopen — CHANGES_REQUESTED** [review]
I was going to approve because 95% of my comments are nits, but there's some commented out code at the end of this PR that disables the welcome guide so I can't approve until that is restored. :)

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue [hunk 3]
```diff
@@ -81,7 +81,7 @@
               <MonitoringPoliciesThresholdCondition
                 v-for="(cond, index) in store.selectedRule!.alertConditions"
                 :key="cond.id"
-                :index="index"
+                :index="index as number"
```
[the remark this answers is not in the document]
Comment (geraldhumphries) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesRuleForm.vue:84]: Index from `v-for` is type `string | number` according to IntelliJ but the component prop is `number`, so it's showing an error. I don't think it had any real impact, implicit type casting was taking care of it. I've changed it to `Number(index)`.

**Comment by mikewrosey** [comment]
As all rules need at least 1 condition with a trigger event, we had this pre-populated with the first option. Should we keep that? If the user attempts to save the policy with the condition as is, it will throw an error.
screenshot-localhost_3009-2023 07 11-12_58_02

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesEventCondition.vue [hunk 4]
```diff
@@ -110,26 +124,19 @@ const severityList = [
...
-  { id: SNMPEventType.SNMP_LINK_DOWN, name: 'SNMP Link Down'},
-]
-
-const triggerEventTypeOptions = [
-  { id: SNMPEventType.SNMP_COLD_START, name: 'SNMP Cold Start' },
-  { id: SNMPEventType.SNMP_WARM_START, name: 'SNMP Warm Start' },
-  { id: SNMPEventType.SNMP_AUTHEN_FAILURE, name: 'SNMP Authentication Failure' },
-  { id: SNMPEventType.SNMP_LINK_DOWN, name: 'SNMP Link Down'},
-  { id: SNMPEventType.SNMP_LINK_UP, name: 'SNMP Link Up'},
-  { id: SNMPEventType.SNMP_EGP_DOWN, name: 'SNMP EGP Down'}
```
Comment (mikewrosey) [inline ui/src/components/MonitoringPolicies/MonitoringPoliciesEventCondition.vue:124]: Looks like this enum `SNMPEventType` is no longer used anywhere. We could remove it entirely from monitoringPolicies.constants.ts

File: ui/src/store/Views/welcomeStore.ts [hunk 5]
```diff
@@ -112,6 +112,7 @@ export const useWelcomeStore = defineStore('welcomeStore', {
       } else {
         router.push('/welcome');
       }
+      onboardingState = false
```
[the remark this answers is not in the document]
Comment (geraldhumphries) [reply ui/src/store/Views/welcomeStore.ts:115]: Oops!

**Comment by geraldhumphries** [comment]
> As all rules need at least 1 condition with a trigger event, we had this pre-populated with the first option. Should we keep that? If the user attempts to save the policy with the condition as is, it will throw an error. screenshot-localhost_3009-2023 07 11-12_58_02



> As all rules need at least 1 condition with a trigger event, we had this pre-populated with the first option. Should we keep that? If the user attempts to save the policy with the condition as is, it will throw an error. screenshot-localhost_3009-2023 07 11-12_58_02

For now I've disabled the save button when there's no triggerEvent on a condition. I think it needs another look, but I suspect the UX will change.

**Review by scthompsonopen — APPROVED** [review]
LGTM, at least the FE portions from a code review standpoint. I can't speak for the BE portions, nor did I manually test this (I'll leave that to those who understand the feature better, I wouldn't know where to start).

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**